src/main/java/org/broadinstitute/hellbender/tools/walkers/mutect/SomaticGenotypingEngine.java

```diff
@@ -189,13 +193,13 @@ public CalledHaplotypes callMutations(
 
     // compute the likelihoods of: AA, AB, AC. . . where A is ref and B, C. . . are somatic alts
     private PerAlleleCollection<Double> getHetGenotypeLogOdds(ReadLikelihoods<Allele> likelihoods,
-                                                              final boolean isNormal,
-                                                              final Strand strand) {
-        final String sample = isNormal ? matchedNormalSampleName : tumorSampleName;
-        final OptionalDouble givenAltAlleleFraction = isNormal ? GERMLINE_HET_ALT_FRACTION : NO_FIXED_TUMOR_ALT_FRACTION;
-        final PerAlleleCollection<Double> hetGenotypeLogLks = getHetGenotypeLogLikelihoods(likelihoods, sample, givenAltAlleleFraction, strand);
+                                                              final boolean isGermline,
+                                                              final Strand strand, final String sample) {
+        final OptionalDouble givenAltAlleleFraction = isGermline ? GERMLINE_HET_ALT_FRACTION : NO_FIXED_TUMOR_ALT_FRACTION;
+        final String sampleForAlleleFractions = (sample == matchedNormalSampleName && isGermline) ? matchedNormalSampleName : tumorSampleName;
```

Comment:
As I understand the code we have 3 use cases of `getHetGenotypeLogOdds`:

1. tumor lod: use the allele fraction of tumor and reads from tumor
2. normal lod: use 0.5 (`GERMLINE_HET_ALT_FRACTION`) as the allele fraction and reads from normal
3. normal artifact lod: use the allele fraction of _tumor_ and reads from normal

With the current code, for case 3) we use the allele fraction of the normal. My understanding is `sampleForAlleleFractions` should always be set to  `tumorSampleName `.